toonz/sources/tnztools/tooloptionscontrols.cpp

```diff
@@ -571,8 +571,12 @@ ToolOptionCombo::ToolOptionCombo(TTool *tool, TEnumProperty *property,
   setSizeAdjustPolicy(QComboBox::AdjustToContents);
   connect(this, SIGNAL(activated(int)), this, SLOT(onActivated(int)));
   // synchronize the state with the same widgets in other tool option bars
-  if (toolHandle)
+  if (toolHandle) {
     connect(this, SIGNAL(activated(int)), toolHandle, SIGNAL(toolChanged()));
+
+    connect(toolHandle, SIGNAL(toolComboBoxListChanged()), this,
+            SLOT(loadEntries()));
```

Comment:
I faced OT crash, when selecting Fill Tool and then selecting Type Tool.

I found that this connection is applied to all combo boxes for now, and once the signal `toolComboBoxList()` is emitted from Type Tool, all the combo boxes which are already constructed invoke `loadEntries()` , even if they are belonging to another tool.

Currently this connection seems to be needed only for the `TypeTool::m_typeFaceMenu` , so you should make it not to be called on other properties.

Reply:
Fixed. I changed how and where that toolComboBoxListChanged signal is set up and processed so is can target a specific combobox on a given tool.